File: api/src/main/java/net/blueshell/api/dto/BlogDTO.java
package net.blueshell.api.dto;


import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;

import java.sql.Timestamp;
import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
@Schema(name = "Blog")
public class BlogDTO extends BaseDTO {
    private Long id;
    private String url;
    @NotBlank
    private String title;
    @NotBlank
    private String html;
    private Instant publishedAt;
}

File: api/src/main/kotlin/net/blueshell/api/dto/BlogDTO.kt
package net.blueshell.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import net.blueshell.api.base.dto.AuditedAutoIdDTO
import java.time.Instant

@Schema(name = "Blog")
data class BlogDTO(
    var url: String? = null,

    @field:NotBlank
    var title: String? = null,

    @field:NotBlank
    var html: String? = null,

    var publishedAt: Instant? = null
) : AuditedAutoIdDTO()
